packages/storybook/src/executors/build-storybook/build-storybook.impl.ts

```diff
@@ -33,7 +33,11 @@ async function runInstance(options: CLIOptions): Promise<void | {
   address: string;
   networkAddress: string;
 }> {
-  const storybookCore = await import('@storybook/core-server');
+  const storybookCore = await (storybookMajorVersion() < 9
```

Comment:
FYI the `@storybook/core-server` -> `storybook/internal/core-server` change was done in Storybook 8.2, and the package won't exist in user's project as long as they don't have it in their package.json. I see that Nx adds it for them, so the package will exist (it does for all Storybook 8 versions).

So in short you can either check for 8.2 and higher to use `storybook/internal/core-server` or just leave this as is, making sure that @storybook/core-server will exist as an explicit dependency